feddlib/core/Mesh/MeshPartitioner_decl.hpp

```diff
@@ -49,43 +49,77 @@ class MeshPartitioner {
     
     ~MeshPartitioner();
     
+	/*! 
+		\brief Main Function of partitioner called 
+	*/
     void readAndPartition();
         
-    /*! Only used in 3D*/
+    /*! \brief Only used in 3D to set the edges as subelements to surfaces*/
     void setEdgesToSurfaces(int meshNumber);
     
+	/*! 
+		\brief Setting surfaces, i.e. edges in 2D and triangles in 3D, as subelements to the corresponding elements
+	*/
     void setSurfacesToElements(int meshNumber);
     
+	/*! 
+		\brief Main function, that reads and partions and distributes the mesh to the different processors. 
+		Here all necessary maps and lists are created
+	*/
     void partitionMesh( MeshUnstrPtr_Type& mesh, int meshNumber );
 
+	/*! 
+		\brief Making the edge list parallel
+	*/
     void buildEdgeListParallel( MeshUnstrPtr_Type mesh, ElementsPtr_Type elementsGlobal );
     
+	/*! 
+		\brief Building the edge list
+	*/
     void buildEdgeList( MeshUnstrPtr_Type mesh, ElementsPtr_Type& elementsGlobal );
 
+	/*! 
+		\brief Setting surfaces, i.e. edges in 2D and triangles in 3D, as subelements to the corresponding elements
+	*/
     void setLocalEdgeIndices(vec2D_int_Type &localEdgeIndices );
 
     void determineRanks();
-    
+
     void determineRanksFromNumberRanks(vec_int_Type& ranks);
     
     void determineRanksFromFractions(vec_int_Type& fractions);
     
     void makeContinuousElements(ElementsPtr_Type elements, vec_idx_Type& eind_vec, vec_idx_Type& eptr_vec );
     
-    void findAndSetSurfaces( vec2D_int_Type& surfElements_vec, vec_int_Type& surfElementsFlag_vec, FiniteElement& element, vec2D_int_Type& permutation, MapConstPtr_Type mapRepeated);
+    //void findAndSetSurfaces( vec2D_int_Type& surfElements_vec, vec_int_Type& surfElementsFlag_vec, FiniteElement& element, vec2D_int_Type& permutation, MapConstPtr_Type mapRepeated);
```

Comment:
Are we going to still need this, or could we remove it?

Reply:
Can be removed.

Reply:
> Is the new test `mesh_nodeFlagsP2` a unit test that prevents this from happening again?

@LeaSassmannshausen  Did you see this one?

Reply:
> Can be removed.

Please do so. When you did all the small changes, we can resolve the comments and push this. Thank you!

Reply:
No. Somehow i am not able to comment on it. Could be build into a unit test at some point. Right now just a very complicated Geometry (flagwise).

Reply:
> No. Somehow i am not able to comment on it. Could be build into a unit test at some point. Right now just a very complicated Geometry (flagwise).

We should do that (at some point).